### ISSUE TITLE
Improve "Incorrect network version" warning message

### ISSUE THREAD

**Issue by Yourname942** [SilentChaos512/Silent-Gear#367]
## Versions

- **Silent Gear**: 1.16.3-2.6.17
- **Silent's Gems**: 1.16.3-3.7.14
- **Silent Lib**: 1.16.3-4.9.6
- **Forge**: 36.1.4
- **Modpack**: https://www.curseforge.com/minecraft/modpacks/the-land-of-super-friends
- **Optifine Installed**: No 

## Expected Behavior

There should be no issues when trying to join the server
(Description here)

## Actual Behavior

I am unable to join the server.

(Description here)

## Links/Images

Image of the error: https://imgur.com/a/vEMtQnO
server log: https://gist.github.com/Yourname942/3e062f19fbdb21f2196ec080463d1bd6
client log: https://gist.github.com/Yourname942/437c4b7796883e9b302c526666d4faa1

## Steps to Reproduce the Problem

1. updated silent gear and silent gems 
2. updated Forge from 36.0.46 to 36.1.4
3. created a new server folder (because of new forge version)
4. ran the forge installer 
5. agreed to Eula/run the server
6. close the server
7. copy + replace the config, default config, world folder, and updated server properties
8 launch the server again
9. run the game again
10. attempt to join server (this image happens)

**Comment by SilentChaos512** [comment]
I thought that message was clear enough, but I guess I need to rewrite
it... You aren't the first to not get it. Anyway, the server you are trying
to connect to is out of date.

**Comment by SilentChaos512** [comment]
Repurposing this issue as a reminder to update the message. Suggestions welcome.

Some possible ways to make the message clearer:
- Add an exception subclass, something like `MismatchedModVersionException`
- Include the mod versions in addition to the network versions
- Shorten the max length of the network version string, so it prints less garbage for outdated servers (like in the screenshot)
- Move the "try updating" sentence to earlier in the message

**Comment by Yourname942** [comment]
> I thought that message was clear enough, but I guess I need to rewrite it... You aren't the first to not get it. Anyway, the server you are trying to connect to is out of date.
> […](#)


How is it out of date if I just downloaded a new Forge installer, created a new server folder, and ran the installer?

**Comment by SilentChaos512** [comment]
@Yourname942 Your server is running a different version of Silent Gear. You likely copied the wrong file. I just tested this and the version detection code isn't broken. I was able to log into a dedicated server running the same version of Silent Gear.

So, double-check your files. I guarantee you that's the issue. Check the file size too to make sure something weird didn't happen with the file names.
